tests/hilti/optimization/unimplemented_hook.hlt

```diff
@@ -0,0 +1,28 @@
+# @TEST-EXEC: hiltic -d -j -D global-optimizer %INPUT > log 2>&1
+# @TEST-EXEC: btest-diff log
+
+# @TEST-EXEC: ${HILTIC} -d -j -g %INPUT -D ast-codegen >ast.0 2>&1
+# @TEST-EXEC: ${HILTIC} -d -j    %INPUT -D ast-codegen >ast.1 2>&1
+# @TEST-EXEC: /usr/bin/env diff -u ast.0 ast.1 | tail -n+3 >output.diff
+# @TEST-EXEC: btest-diff output.diff
+
+# @TEST-DOC: Tests optimizations removing unimplemented hooks.
```

Comment:
Is the optimization also handling global hooks (outside of struct) already? If so, add a test for that as well (and if not, something for the TODO list to support as well).

Reply:
This is currently not handled. I added a `TODO`.